feat(marker_radar_lidar_calibrator): marker_radar_lidar_calibrator support for different radar msgs and transformation algorithms

vividf opened the pull request:
## Description
Same description as https://github.com/tier4/CalibrationTools/pull/161 after rebasing on the documentation branch. 

Our previous tools assumed that radar has no elevation and we provided two algorithms:

- Yaw-only calibration
- x,y, yaw calibration

With elevation, we can get the full 6D pose. However, some sensors may impose restrictions on the orientation. In particular, the ARS548 imposes roll=0 for its object interface. We need an algorithm that easily accommodates this new restriction.

With this PR, we provide four algorithms

- Yaw-only calibration (yaw_only_rotation_2d)
- x,y, yaw calibration (svd_2d)
- 3d calibration (svd_3d)
- 3d transformation with roll is restricted to zero (roll_zero_3d)

and three different input msg-type for radar

- radar_tracks
- radar_scan
- radar_cloud

Additionally, this PR also includes the xx1 gen2 project in the sensor_calibration_manager.

<!-- Write a brief description of this PR. -->

## Related links

<!-- Write the links related to this PR. -->

## Tests performed

<!-- Describe how you have tested this PR. -->

## Notes for reviewers

<!-- Write additional information if necessary. It should be written if there are related PRs that should be merged at the same time. -->

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [ ] I've confirmed the [contribution guidelines].
- [ ] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

vividf committed:
changed the name from radar_parallel to radar_optimization, x2 output changed to parent_to_radar

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
integrate with x2, still need to adapt to other

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
add comment for x2 project

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
fix spell check

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
fix spell error!

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
simplify cmake

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
fix the residuals from 4d to 3d

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
fix variable name and add comment

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
fix initial calibration error explanation

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
add BEV enviroment

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
add vis image for mapping based

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
change to svg file

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
fix errors

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
add resolution explaination

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
fix lidar to camera

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
remove last point

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
seperate the context

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
remove accurately

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
remove addtional the

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
add link

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
fix message to object and add reason of reliably detecting challenge

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
fix section 2

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
improve step 3

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
fix calibration camera

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
add docker solution

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
fix some sentences in mapping_based

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
fix the input parameters

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

pre-commit-ci[bot] committed:
ci(pre-commit): autofix

knzo25 committed:
chore: applied changes from the auto messages migration

note: this was note made as another PR since it also involved documentation

Signed-off-by: Kenzo Lobos-Tsunekawa <kenzo.lobos@tier4.jp>

knzo25 committed:
chore: mispells

Signed-off-by: Kenzo Lobos-Tsunekawa <kenzo.lobos@tier4.jp>

knzo25 committed:
chore: another attempt to fix the mispell

Signed-off-by: Kenzo Lobos-Tsunekawa <kenzo.lobos@tier4.jp>

knzo25 committed:
chore: forgot to commit one .repos file

Signed-off-by: Kenzo Lobos-Tsunekawa <kenzo.lobos@tier4.jp>

knzo25 committed:
chore: replaced private links for public ones

Signed-off-by: Kenzo Lobos-Tsunekawa <kenzo.lobos@tier4.jp>

vividf committed:
chore: fix input

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: fix the FAQ

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: fix radar purpose

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
predict to estimate

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: fix radar vis image

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: fix foreground extraction

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: fix radar step 3 explaination

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: fix add_lidar_radar_pair parameter explanation

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: fix radar Pro tips/recommendations

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: fix radar foreground pointcloud

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: replace the pnp lidar pos to correct one

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: fix didn't to did not

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: fix tag pos explanation

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: remove some text

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: fix some sentence of mapping

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: add radar foreground and background image and explanation

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: fix parameter desciption

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: fix larger to exceed

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: fix grammar errors in mapping based calibrator

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: fix grammar errors in marker based calibrator

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

knzo25 committed:
chore: fixed ci/cd

Signed-off-by: Kenzo Lobos-Tsunekawa <kenzo.lobos@tier4.jp>

knzo25 committed:
chore: ci/cd part 2

Signed-off-by: Kenzo Lobos-Tsunekawa <kenzo.lobos@tier4.jp>

knzo25 committed:
chore: moved meta package's path

Signed-off-by: Kenzo Lobos-Tsunekawa <kenzo.lobos@tier4.jp>

knzo25 committed:
chore: changed private repo to oss one

Signed-off-by: Kenzo Lobos-Tsunekawa <kenzo.lobos@tier4.jp>

vividf committed:
chore: Update calibrators/marker_radar_lidar_calibrator/README.md

Co-authored-by: Kenzo Lobos Tsunekawa <kenzo.lobos@tier4.jp>

vividf committed:
chore: fix marker calibrator algorithms explanation

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: add calibration process

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: remove first

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: fix grammar

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: remove finalize

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: fix Z to z

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: specify yaw rotation

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: fix all of the explanation for output parameters

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: change predictable

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: remove redundant calibration area

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
fix pro tips

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: change to calibrating

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: fix more grammar errors

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: add more info for the radar foreground image

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: add explanation for estimation algorithm

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: add svd explanation

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: fix radar reflector explanation

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: remove while

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: remove angle

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: fix mapping errors

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: update calibrators/marker_radar_lidar_calibrator/README.md

Co-authored-by: Kenzo Lobos Tsunekawa <kenzo.lobos@tier4.jp>

vividf committed:
chore: fix grammar

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: fix radar pro tips

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
Merge branch 'feature/new_api_documentation' of github.com:tier4/CalibrationTools into HEAD

vividf committed:
chore: update calibrators/tag_based_pnp_calibrator/README.md

Co-authored-by: Kenzo Lobos Tsunekawa <kenzo.lobos@tier4.jp>

vividf committed:
chore: add pnp known issues

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: fix grammar error, pnp overall calibration env

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: fix grammar error: that

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: fix pnp FAQ

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: add the

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: reflectors's tracks

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: build to mount

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: fix final step

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: add comma

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
fix: fix radar foreground points to object

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: remove that is

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: fix denote as

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: add explanation for step 4

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: fix services

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: fix radar parallel frame

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: fix transformation type

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: add unit1

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: add unit2

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: no reflector, person, ...

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore:  imperfect to inaccuarate

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: fix object_raw

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: fix and to or

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: fix show like sentence

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: fix link, not in comment

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: add explanation for radar foreground

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: fix add lidar-radar pair

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: fix blue point explanation

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: explain two radar , one lidar

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: line to curve

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: change coordinate to frame

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: add explanation for ground segmentation in radar-lidar calibrator

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: add ground segmentation model

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

knzo25 committed:
chore: fixed compilations issues due to changes in universe

Signed-off-by: Kenzo Lobos-Tsunekawa <kenzo.lobos@tier4.jp>

knzo25 committed:
chore: colcon ignored the tunable tf package since it produces build errors and is not being maintained

Signed-off-by: Kenzo Lobos-Tsunekawa <kenzo.lobos@tier4.jp>

knzo25 committed:
Merge branch 'tier4/universe' into feature/new_api_documentation

knzo25 committed:
chore: attepting to update ci/cd since it seemed old and it was failing

Signed-off-by: Kenzo Lobos-Tsunekawa <kenzo.lobos@tier4.jp>

knzo25 committed:
chore: removed galactic from ci/cd since it is unsupported and was still part of our ci/cd

Signed-off-by: Kenzo Lobos-Tsunekawa <kenzo.lobos@tier4.jp>

knzo25 committed:
chore: fixed missing cpell ignores

Signed-off-by: Kenzo Lobos-Tsunekawa <kenzo.lobos@tier4.jp>

vividf committed:
chore: fix mapping calibrator

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
fix: fix spell check

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
fix: fix grammar errors

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
fix: fix mapping section

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
fix: fix errors

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
fix: fix grammar error

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
fix: fix parameters spell check

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

pre-commit-ci[bot] committed:
ci(pre-commit): autoupdate (#174)

updates:
- [github.com/pre-commit/mirrors-clang-format: v18.1.5 → v18.1.7](https://github.com/pre-commit/mirrors-clang-format/compare/v18.1.5...v18.1.7)

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

vividf committed:
fix: fix transformation optimization and add support for radar input pointcloud

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: remove the redundant line

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: fix central to center

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: add between sensors

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: fix grids to grid

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: fix transformation type explanation

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: fix rigid transformation estimation

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: add additional explanation

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: remove processes

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: change pinpoint to estiamte

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: fix topic name

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: fix doesn't to does not

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: add more explanation to radar parallel frame

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: fix topic

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: unify radar parallel frame and clarify z component

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
Update calibrators/docs/tutorials/marker_radar_lidar_calibrator.md

Co-authored-by: Kenzo Lobos Tsunekawa <kenzo.lobos@tier4.jp>

vividf committed:
Update calibrators/docs/tutorials/tag_based_pnp_calibrator.md

Co-authored-by: Kenzo Lobos Tsunekawa <kenzo.lobos@tier4.jp>

vividf committed:
Update calibrators/docs/tutorials/tag_based_pnp_calibrator.md

Co-authored-by: Kenzo Lobos Tsunekawa <kenzo.lobos@tier4.jp>

vividf committed:
Update calibrators/marker_radar_lidar_calibrator/README.md

Co-authored-by: Kenzo Lobos Tsunekawa <kenzo.lobos@tier4.jp>

vividf committed:
chore: fix grammar error

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: fix at which

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: add in seconds unit

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: fix max_no_observation_time

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: fix new_hypothesis_distance

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: fix to degrees

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: fix all square issues

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: fix lowercase

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: ARS408 radars

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: add missing the

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: fix input_lidar_pointcloud

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: fix layout

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: fix more layout

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: fix sentences

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: fix in rviz

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: fix grammar error: of

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: move butoon out

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: radar_foreground_pointcloud

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: remove quotation marks

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
fix: fix mapping based

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: fix more grammar error

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: remove almost deprecated function

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
feat: use template for radar input msg

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
Update calibrators/marker_radar_lidar_calibrator/README.md

Co-authored-by: Kenzo Lobos Tsunekawa <kenzo.lobos@tier4.jp>

vividf committed:
chore: fix all rviz

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: fix unit

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: fix detections

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: fix ARS

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: fix like below

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: fix uppercase

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: fix context in mapping

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: remove some text

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: fix some grammar

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: merge docs branch

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
Update calibrators/mapping_based_calibrator/README.md

Co-authored-by: Kenzo Lobos Tsunekawa <kenzo.lobos@tier4.jp>

vividf committed:
Update calibrators/mapping_based_calibrator/README.md

Co-authored-by: Kenzo Lobos Tsunekawa <kenzo.lobos@tier4.jp>

vividf committed:
Update calibrators/mapping_based_calibrator/README.md

Co-authored-by: Kenzo Lobos Tsunekawa <kenzo.lobos@tier4.jp>

vividf committed:
chore: fix spell error

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
Update calibrators/mapping_based_calibrator/README.md

Co-authored-by: Kenzo Lobos Tsunekawa <kenzo.lobos@tier4.jp>

vividf committed:
chore: fix link

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: fix known issues grammar

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: known issues grammar error2

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: tips for distortion

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: fix all base_link

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: fix step 2

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: dense_pointcloud_num_keyframes plural fix

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: for developers

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: keyframe markers and the

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: extracted from

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: remove ()

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: add pro-tips

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: fix step 2 calibration data preparation

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: add processing

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: fix data selection

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: fix more data selection

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: would be to identical

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: could get to obtain

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: rebase to documentation branch

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

pre-commit-ci[bot] committed:
ci(pre-commit): autofix

vividf commented:
@knzo25 
If I create a PR for merging into the documentation file, only 30? changed files will show. 
I saw that you changed the base in the previous PR, does that help?

![image](https://github.com/user-attachments/assets/02547e3c-076d-429c-8f7c-caf28f3b0097)

pre-commit-ci[bot] committed:
ci(pre-commit): autoupdate (#176)

updates:
- [github.com/tier4/pre-commit-hooks-ros: v0.8.0 → v0.10.0](https://github.com/tier4/pre-commit-hooks-ros/compare/v0.8.0...v0.10.0)
- [github.com/pre-commit/mirrors-clang-format: v18.1.7 → v18.1.8](https://github.com/pre-commit/mirrors-clang-format/compare/v18.1.7...v18.1.8)

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

knzo25 committed:
fix: fixed dependencies (due to autoware's autoware prefix)

Signed-off-by: Kenzo Lobos-Tsunekawa <kenzo.lobos@tier4.jp>

knzo25 committed:
chore: moved the sensor calibrator manager outside of the calibrators folder (other changes are pre-commit induced)

Signed-off-by: Kenzo Lobos-Tsunekawa <kenzo.lobos@tier4.jp>

knzo25 committed:
chore: updated the use of the kalman filter implementation since it was moved into the autoware namespace

Signed-off-by: Kenzo Lobos-Tsunekawa <kenzo.lobos@tier4.jp>

vividf committed:
chore: add option in rdv and x2 launcher

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

knzo25 committed:
fix: updated the use of the decompressor

Signed-off-by: Kenzo Lobos-Tsunekawa <kenzo.lobos@tier4.jp>

pre-commit-ci[bot] committed:
ci(pre-commit): autoupdate (#184)

updates:
- [github.com/scop/pre-commit-shfmt: v3.8.0-1 → v3.9.0-1](https://github.com/scop/pre-commit-shfmt/compare/v3.8.0-1...v3.9.0-1)
- [github.com/psf/black: 24.4.2 → 24.8.0](https://github.com/psf/black/compare/24.4.2...24.8.0)

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

knzo25 committed:
chore: adapted to the new way of using lidartag

Signed-off-by: Kenzo Lobos-Tsunekawa <kenzo.lobos@tier4.jp>

vividf committed:
fix: init tf tree visualization (#188)

* fix: init tf tree visualization

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

* chore: fix error

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

---------

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

knzo25 committed:
fix(intrinsic_camera_calibrator): fixed dot board detector (#189)

fix: forgot to add a fix to the dot board detector

Signed-off-by: Kenzo Lobos-Tsunekawa <kenzo.lobos@tier4.jp>

knzo25 committed:
docs: candidate for the final documentation of the new api

Signed-off-by: Kenzo Lobos-Tsunekawa <kenzo.lobos@tier4.jp>

knzo25 committed:
chore: fixed dead links and spelling

Signed-off-by: Kenzo Lobos-Tsunekawa <kenzo.lobos@tier4.jp>

knzo25 committed:
chore: fixed missing mispell in a package name

Signed-off-by: Kenzo Lobos-Tsunekawa <kenzo.lobos@tier4.jp>

knzo25 committed:
chore: fixed missing dead link

Signed-off-by: Kenzo Lobos-Tsunekawa <kenzo.lobos@tier4.jp>

knzo25 commented:
@vividf 
Please rebase this branch and the one with your hacks for the garage calibration ASAP

vividf committed:
chore: solve conflicts

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: merge tier4/main to feature branch

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

pre-commit-ci[bot] committed:
ci(pre-commit): autofix

vividf committed:
chore: fix spell error

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: fix directory

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: fix docs

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf commented:
@knzo25 Please use this branch for garage calibration https://github.com/vividf/CalibrationTools/tree/t4_garage_xx1_experiment

vividf committed:
chore: fix package dependency

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

knzo25 commented:
@vividf 
Please remove all non radar-lidar calibration changes (XX1 Gen2 should be in its own PR. Said PR should not include the changes for the radar-lidar)

knzo25 commented:
@vividf 
Check the comments at https://github.com/tier4/CalibrationTools/pull/195
I can not make a proper functional evaluation yet due to the high numbers of false positives.

I will make some code-related comments later, but in the meantime:
 - due to the nature of the new algorithms, visualization among different optimization models is necessary. For example, to compare how much accuracy we are sacrificing due to the optimization restrictions (rotation vs 2d svd. no-roll vs 3d svd)
 - Since new radars will probably will increase the false positive rate anyways, instead of removing the last detection we need a `remove calibration pair i`, with the index number being displayed in RViz

knzo25 commented on calibrators/marker_radar_lidar_calibrator/include/marker_radar_lidar_calibrator/marker_radar_lidar_calibrator.hpp:
```diff
@@ -119,19 +132,21 @@ class ExtrinsicReflectorBasedCalibrator : public rclcpp::Node
-  void crossValEvaluation(
-    pcl::PointCloud<PointType>::Ptr lidar_points_pcs,
-    pcl::PointCloud<PointType>::Ptr radar_points_rcs);
+  void selectCombinations(
```
nit: consts and adequate size types (we did not pay attention before, but we are trying now)

knzo25 replied:
This one may have several implications across the codebase

vividf replied:
Thanks, fixed in https://github.com/tier4/CalibrationTools/pull/180/commits/b9e8c9d0e3d1cc8afdb5fda5f29a6738e3e13ebb

knzo25 commented on calibrators/marker_radar_lidar_calibrator/include/marker_radar_lidar_calibrator/marker_radar_lidar_calibrator.hpp:
```diff
@@ -119,19 +132,21 @@ class ExtrinsicReflectorBasedCalibrator : public rclcpp::Node
-    pcl::PointCloud<PointType>::Ptr radar_points_rcs);
+  void selectCombinations(
+    int tracks_size, int num_of_samples, std::vector<std::vector<int>> & combinations);
+  void doEvaluation(std::vector<std::vector<int>> & combinations, int num_of_samples);
```
doEvaluation -> evaluate ?

knzo25 replied:
In context: evaluateCombinations?

vividf replied:
Thanks, fixed in https://github.com/tier4/CalibrationTools/pull/180/commits/2d0accd9bddda43bf71d4b1e15e6ef9d1be049bc

knzo25 commented on calibrators/marker_radar_lidar_calibrator/include/marker_radar_lidar_calibrator/marker_radar_lidar_calibrator.hpp:
```diff
@@ -119,19 +132,21 @@ class ExtrinsicReflectorBasedCalibrator : public rclcpp::Node
-  std::tuple<pcl::PointCloud<PointType>::Ptr, pcl::PointCloud<PointType>::Ptr, double, double>
-  getPointsSetAndDelta();
+  std::tuple<pcl::PointCloud<PointType>::Ptr, pcl::PointCloud<PointType>::Ptr> getPointsSet();
+  std::tuple<double, double> getDelta(std::vector<Track> converged_tracks, bool is_crossval);
```
Please use a more descriptive name
getRotationDelta
get2DRotationDelta
etc

vividf replied:
Thanks, fixed in https://github.com/tier4/CalibrationTools/pull/180/commits/0f648c0f7e2ef653b4c604e4155bc7d36cc7c093

knzo25 commented on calibrators/marker_radar_lidar_calibrator/include/marker_radar_lidar_calibrator/marker_radar_lidar_calibrator.hpp:
```diff
@@ -149,8 +164,9 @@ class ExtrinsicReflectorBasedCalibrator : public rclcpp::Node
   {
-    std::string radar_parallel_frame;  // frame that is assumed to be parallel to the radar (needed
-                                       // for radars that do not provide elevation)
+    std::string radar_optimization_frame;  // frame that is assumed to be parallel to the radar
```
I think that this is not true anymore and as such the description should change.

For example, `base_link` will be used as the parallel frame when calibrating the ARS548 right?

knzo25 commented on calibrators/marker_radar_lidar_calibrator/include/marker_radar_lidar_calibrator/marker_radar_lidar_calibrator.hpp:
```diff
@@ -149,8 +164,9 @@ class ExtrinsicReflectorBasedCalibrator : public rclcpp::Node
   {
-    std::string radar_parallel_frame;  // frame that is assumed to be parallel to the radar (needed
-                                       // for radars that do not provide elevation)
+    std::string radar_optimization_frame;  // frame that is assumed to be parallel to the radar
+                                           // if estimating the transformation by 2d algorithms
+                                           // (needed for radars that do not provide elevation)
```
I think that this is not true anymore and as such the description should change.

For example, base_link will be used as the parallel frame when calibrating the ARS548 right?

vividf replied:
Thanks, I rewrote the description in https://github.com/tier4/CalibrationTools/pull/180/commits/b2e2da78c451a706c76f9d02ceabf4d3b1b019fb

knzo25 commented on calibrators/marker_radar_lidar_calibrator/include/marker_radar_lidar_calibrator/marker_radar_lidar_calibrator.hpp:
```diff
@@ -233,8 +251,12 @@ class ExtrinsicReflectorBasedCalibrator : public rclcpp::Node
+  geometry_msgs::msg::Transform radar_optimization_to_lidar_msg_;
+  Eigen::Isometry3d radar_optimization_to_lidar_eigen_;
+
+  geometry_msgs::msg::Transform init_radar_optimization_to_radar_msg_;
```
There are some variables with `init` and others with `initial`

vividf replied:
Thanks, fixed in https://github.com/tier4/CalibrationTools/pull/180/commits/dfc339d87a90bff50e6cd5c4ae6facffdd39355b

knzo25 commented on calibrators/marker_radar_lidar_calibrator/include/marker_radar_lidar_calibrator/marker_radar_lidar_calibrator.hpp:
```diff
@@ -263,9 +287,15 @@ class ExtrinsicReflectorBasedCalibrator : public rclcpp::Node
   std::vector<Track> converged_tracks_;
 
+  // Converged points
+  pcl::PointCloud<PointType>::Ptr lidar_points_ocs_;
```
In the following code:

```
output_metrics_.clear();
estimateTransformation();
crossValEvaluation();
publishMetrics();
```

`lidar_points_ocs_` and its radar counterpart are filled and used inside a method inside `estimateTransformation` and used in another method inside `crossValEvaluation`.

When the code block in question (create pairs using `getPointsSet`) is not that heave and not kept between "iterations" it improves readability not using method variables

vividf replied:
Thanks! refactored a bit in https://github.com/tier4/CalibrationTools/pull/180/commits/a62aae70e5cafec37f3d28a9270e65a17b4e33c4

knzo25 commented on calibrators/marker_radar_lidar_calibrator/include/marker_radar_lidar_calibrator/transformation_estimator.hpp:
```diff
@@ -0,0 +1,66 @@
+class TransformationEstimator
+{
+public:
+  using PointType = pcl::PointXYZ;
```
This error was before but can you make all instances of PointType be declared only once ?
(currently it is defined three times, and there is the risk of changing one and forgetting the others - quite low though)

vividf replied:
fixed in https://github.com/tier4/CalibrationTools/pull/180/commits/002d87c59f4884162e45abcbd9668ea9258edb68

knzo25 commented on calibrators/marker_radar_lidar_calibrator/include/marker_radar_lidar_calibrator/transformation_estimator.hpp:
```diff
@@ -0,0 +1,66 @@
+  void setPoints(
+    pcl::PointCloud<PointType>::Ptr lidar_points_ocs,
+    pcl::PointCloud<PointType>::Ptr radar_points_rcs);
+  void setDelta(double delta_cos, double delta_sin);
```
Commented before, but this name could be improved

vividf replied:
Thanks, fixed in https://github.com/tier4/CalibrationTools/pull/180/commits/7f7f8e57197a8affd7d81f2d375adf2fc1fcc5f8

knzo25 commented on calibrators/marker_radar_lidar_calibrator/src/marker_radar_lidar_calibrator.cpp:
```diff
@@ -365,7 +404,7 @@ void ExtrinsicReflectorBasedCalibrator::backgroundModelRequestCallback(
 
     RCLCPP_WARN_THROTTLE(
-      this->get_logger(), *this->get_clock(), 5000, "Waiting for the calibration to end");
+      this->get_logger(), *this->get_clock(), 5000, "Waiting for the extracting background model");
```
Grammar

vividf replied:
Thanks, fixed in https://github.com/tier4/CalibrationTools/pull/180/commits/ba05cc4ce7421b28de27fb995607aed1de5c3516

vividf replied:
Thanks, fixed in https://github.com/tier4/CalibrationTools/pull/180/commits/ba05cc4ce7421b28de27fb995607aed1de5c3516

knzo25 commented on calibrators/marker_radar_lidar_calibrator/src/marker_radar_lidar_calibrator.cpp:
```diff
@@ -434,15 +473,37 @@ void ExtrinsicReflectorBasedCalibrator::lidarCallback(
+    latest_radar_scan_msgs_->returns.clear();
+  } else {
+    if (!latest_radar_cloud_msgs_) {
+      RCLCPP_INFO(this->get_logger(), "There were no radar pointcloud");
```
grammar

knzo25 commented on calibrators/marker_radar_lidar_calibrator/src/marker_radar_lidar_calibrator.cpp:
```diff
@@ -434,15 +473,37 @@ void ExtrinsicReflectorBasedCalibrator::lidarCallback(
+  } else if (msg_type_ == MsgType::radar_scan) {
+    if (!latest_radar_scan_msgs_ || latest_radar_scan_msgs_->returns.size() == 0) {
+      if (latest_radar_scan_msgs_->returns.size() == 0)
+        RCLCPP_INFO(this->get_logger(), "There were no scan");
```
grammar

vividf replied:
Addressed this one and above ones in https://github.com/tier4/CalibrationTools/pull/180/commits/e5570b8e954d779a83bd92ac2520221349da4fcc

knzo25 commented on calibrators/marker_radar_lidar_calibrator/src/marker_radar_lidar_calibrator.cpp:
```diff
@@ -459,19 +520,37 @@ void ExtrinsicReflectorBasedCalibrator::lidarCallback(
 {
-  if (!latest_radar_msgs_) {
-    latest_radar_msgs_ = msg;
+  if (!latest_radar_tracks_msgs_) {
```
Regarding the methods `radarTrackscallback` and `radarScanCallback`.

At the beginning of X2 Gen1, we has driver issues and the frequency of the radar was dozen of times higher than the lidars.

This caused the need to aggregate the radar scans and make the calibration radar-centric (i.e., calibrate each lidar scan).

This error was fixed somewhere along the line and now the rosbags look like this:
```
Topic information: Topic: /sensing/lidar/front_lower/pointcloud_raw | Type: sensor_msgs/msg/PointCloud2 | Count: 2064 | Serialization Format: cdr
                   Topic: /sensing/radar/front_center/objects_raw | Type: radar_msgs/msg/RadarTracks | Count: 2738 | Serialization Format: cdr
                   Topic: /tf_static | Type: tf2_msgs/msg/TFMessage | Count: 1 | Serialization Format: cdr
```

i.e., almost the same frec.
I think we can safely remove this approach now

vividf replied:
@knzo25 Do you mean like this?
https://github.com/tier4/CalibrationTools/pull/180/commits/8ab4a0c2fffa75abdd7d8ec0d921748dfa2bc85f

vividf replied:
fix: https://github.com/tier4/CalibrationTools/pull/180/commits/8ab4a0c2fffa75abdd7d8ec0d921748dfa2bc85f
revert: https://github.com/tier4/CalibrationTools/pull/180/commits/b6c3c69c4076575c97d61f3da5d7bae5ae4c1567

I reverted this fix since it causes a problem on x2 gen2 calibration.
I will fix this later.

knzo25 commented on calibrators/marker_radar_lidar_calibrator/src/marker_radar_lidar_calibrator.cpp:
```diff
@@ -587,11 +666,48 @@ std::vector<Eigen::Vector3d> ExtrinsicReflectorBasedCalibrator::extractReflector
+  } else if constexpr (std::is_same<RadarMsgType, radar_msgs::msg::RadarScan>::value) {
+    radar_pointcloud_ptr->reserve(msg->returns.size());
+    for (const auto & radar_return : msg->returns) {
+      float range = radar_return.range;
```
kind of nit:
you are creating copies. whenever applicable use const references

knzo25 commented on calibrators/marker_radar_lidar_calibrator/src/marker_radar_lidar_calibrator.cpp:
```diff
@@ -1249,134 +1398,107 @@ std::pair<double, double> ExtrinsicReflectorBasedCalibrator::computeCalibrationE
+  calculateCalibrationError(calibrated_radar_to_lidar_transformation);
+}
 
+void ExtrinsicReflectorBasedCalibrator::calculateCalibrationError(
```
`calculateCalibrationError` and `computeCalibrationError` exist in this file with similar yet different purposes and signatures. This makes it very difficult to read. Choose one and overload, or use different more appropriate words

knzo25 replied:
The current `calculateCalibrationError` is unnecessarily deep in the call hierarchy.
In code metrics, the less deep the better.

Semantically speaking it should be between:
```
estimateTransformation();
<-- 
crossValEvaluation();
```

vividf replied:
Thanks! fixed in https://github.com/tier4/CalibrationTools/pull/180/commits/c1a1ea3e4ea4d90e057ab97497431000437e3acb

knzo25 commented on calibrators/marker_radar_lidar_calibrator/src/marker_radar_lidar_calibrator.cpp:
```diff
@@ -1249,134 +1398,107 @@ std::pair<double, double> ExtrinsicReflectorBasedCalibrator::computeCalibrationE
-      "The pure rotation calibration pose was chosen as the output calibration pose. This may mean "
-      "you need to collect more points");
-    calibrated_radar_to_lidar_eigen_ = calibrated_rotation_radar_to_lidar_transformation;
+      this->get_logger(), "The calibration pose was chosen as the output calibration pose");
```
The output messages in this method make sense in the context of the code before this PR. Now they do not make much sense. Please fix them

vividf replied:
fixed in https://github.com/tier4/CalibrationTools/pull/180/commits/9bee4d290ba74eea1a9ccc7c45a7a745f2954e78

knzo25 commented on calibrators/marker_radar_lidar_calibrator/src/marker_radar_lidar_calibrator.cpp:
```diff
@@ -1402,108 +1524,124 @@ void ExtrinsicReflectorBasedCalibrator::findCombinations(
+
+void ExtrinsicReflectorBasedCalibrator::crossValEvaluation()
+{
+  int tracks_size = static_cast<int>(converged_tracks_.size());
```
can you change the parts that correspond to std::size?

knzo25 replied:
(this applies in several parts of the crossvalidation strategy )

vividf replied:
Addressed them in https://github.com/tier4/CalibrationTools/pull/180/commits/b9e8c9d0e3d1cc8afdb5fda5f29a6738e3e13ebb

knzo25 commented:
@vividf 
Everywhere in the code that a `.z() = 0` exists, it is only applicable to the old 2D versions of the algorithm. This one is an important error, and any result that comes with it has a non negligible error
I was worried that the error that I saw using this tool was too high. Hopefully this one fixes it...

knzo25 commented on calibrators/marker_radar_lidar_calibrator/src/transformation_estimator.cpp:
```diff
@@ -0,0 +1,188 @@
+{
+  RCLCPP_INFO(
+    rclcpp::get_logger("marker_radar_lidar_calibrator"),
+    "Estimate 3D transformation with roll is always zero");
```
Is this a statement ?!?!?!

vividf replied:
Changed the logging to https://github.com/tier4/CalibrationTools/pull/180/commits/12b356233707aa31d5be2cd629d19bbcef3401cb

knzo25 commented on calibrators/marker_radar_lidar_calibrator/src/transformation_estimator.cpp:
```diff
@@ -0,0 +1,188 @@
+  RCLCPP_INFO(
+    rclcpp::get_logger("marker_radar_lidar_calibrator"), "%s", initial_params_msg.c_str());
+
+  for (size_t i = 0; i < lidar_points_ocs_->points.size(); i++) {
```
We have been trying to use std::size_t so far

vividf replied:
addressed in https://github.com/tier4/CalibrationTools/pull/180/commits/28fdcf92c92194da78cae2678541c7a4a1e62b92

knzo25 commented on calibrators/marker_radar_lidar_calibrator/src/transformation_estimator.cpp:
```diff
@@ -0,0 +1,188 @@
+
+    ceres::CostFunction * cost_function = new ceres::AutoDiffCostFunction<SensorResidual, 3, 5>(
+      new SensorResidual(radar_point_eigen, lidar_point_eigen));
+    problem.AddResidualBlock(cost_function, NULL, params.data());
```
nullptr

vividf replied:
Thanks, fixed in https://github.com/tier4/CalibrationTools/pull/180/commits/ef7043564d408baaaeb37eb085908c4e24a58333

knzo25 reviewed (changes requested):
Left several comments. The algorithm is solid, but this PR still needs quite a lot of work.
When I have a commit more stable to do some evaluations I will probably have more comments, but I will wait until the current comments have been addressed

vividf committed:
chore: fix default rviz for text markers

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: remove unused header

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: change function name to evaluateCombinations

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: change function name to get2DRotationDelta

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: change variable name

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: fix waiting to extract the background model

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: fix grammar error

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: fix function name and clean code

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: remove unnessary logging

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: fix int to size_t

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: fix estimateZeroRollTransformation logging

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: add std::size_t

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: NULL to nullptr

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: declare point type only once

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: remove class member and declare new structure

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: fix description of radar_optimization_frame

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: remove if condition

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
chore: fix function name setDelta

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
Revert "chore: remove if condition"

This reverts commit 8ab4a0c2fffa75abdd7d8ec0d921748dfa2bc85f.

vividf committed:
chore: fix bug

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

knzo25 commented:
@vividf 
Was this addressed? If so can you tell me the commit?

>@vividf
Everywhere in the code that a .z() = 0 exists, it is only applicable to the old 2D versions of the algorithm. This one is an important error, and any result that comes with it has a non negligible error
I was worried that the error that I saw using this tool was too high. Hopefully this one fixes it...

vividf committed:
chore: fix yaw error

Signed-off-by: vividf <yihsiang.fang@tier4.jp>

vividf committed:
Merge branch 'tier4/universe' into feature/marker_radar_lidar_calibrator_support_radars_and_transformation_algorithms

vividf commented:
@knzo25 
Before I fixed it in https://github.com/tier4/CalibrationTools/pull/180/commits/a62aae70e5cafec37f3d28a9270e65a17b4e33c4 but seems that error is still high. 

I just found out that I didn't fix the yaw error function so I just fixed it in https://github.com/tier4/CalibrationTools/pull/180/commits/64d8e53cff321e9da3a37795262059a535b52d1b.

The error is a bit lower than before but still high (I will send PM to you on slack)

knzo25 commented:
Thanks, just now I am catching up on the radar calibration, to make  list of the next steps

knzo25 commented:
@vividf 
 - pre-commit does not pass
 - This PR should be rebased if possible
 - XX1 Gen2 unrelated to the radar should be in another PR
 - Since we are going to (probably) end up restricting the calibration FOV with a cone, it should both put as a marker (guideline) and also not accept samples outside
 -  (Calibrator data) Variance indicators will be vital, since using  single value of elevation in the reflectors end up in a degenerate optimization problem (for example, in the camera-lidar, each tag has 4 points, so the issue does not arise)
 - We had talked before, but we need a button to delete a reflector by index. In the same way, it would be convenient to have a cancel a pair addition.
 - Camera-lidar, base-lidar, and lidar-lidar have ways to save the data used for calibration for offline analysis. Please add an option (potentially a service), to save the calibration points
 - The radar markers were implemented as a line with a square, representing the fact that we do not know the z component of RadarTrack (ARS408). The same is not true for 4D radars (with elevation). Please propose a new way to visualize it

knzo25 commented on sensor_calibration_manager/launch/default_project/marker_radar_lidar_calibrator.launch.xml:
```diff
@@ -2,11 +2,13 @@
+  <arg name="radar_optimization_frame"/>
   <arg name="radar_frame"/>
   <arg name="lidar_frame"/>
+  <arg name="msg_type"/>
```
for msg_type and transformation_type, please provide alternatives with the choice attribute

vividf replied:
Fixed in the new PR https://github.com/tier4/CalibrationTools/pull/221

knzo25 commented on calibrators/marker_radar_lidar_calibrator/launch/calibrator.launch.xml:
```diff
@@ -4,9 +4,11 @@
   <arg name="input_lidar_pointcloud" default="/sensing/lidar/front_lower/pointcloud_raw"/>
-  <arg name="input_radar_objects" default="/sensing/radar/front_center/objects_raw"/>
+  <arg name="input_radar_msg" default="/sensing/radar/front_center/objects_raw"/>
+  <arg name="msg_type" default="objects_raw"/>
```
for msg_type and transformation_type, please provide alternatives with the choice attribute

vividf replied:
Fixed in the new PR https://github.com/tier4/CalibrationTools/pull/221

knzo25 reviewed (changes requested):
Checked calibration with X2Gen2 data.
Made general comments since the results are not near the required level yet.
Will be updating tickets later

vividf commented:
@knzo25 
Thanks for your review!
Since this PR is too complicated to rebase after the documentation branch merges into the main branch, I have opened another PR that includes most of the new features you mentioned above. Most of the requested features are implemented, except for the cone-related one, which I will address after PR [#221](https://github.com/tier4/CalibrationTools/pull/221) is merged.
